### ISSUE TITLE
Persistent Wind Intensity Patterns in Historical Typhoon Tracks Beyond 180 Degrees Longitude

### ISSUE THREAD

**Issue by fedemoss** [CLIMADA-project/climada_python#831]
**Issue Description:**

I'm encountering consistent patterns while plotting wind intensity for historical typhoons that occurred in Fiji using CLIMADA v4.0.1. I'm using the TropCyclone class from climada.hazard and extracting tracks from specific Fiji historical typhoons via `TCTracks.from_ibtracs_netcdf()`.

**Problem Statement:**

The issue arises consistently when dealing with typhoon tracks where longitudes exceed 180 degrees. Despite the typhoons occurring in different years, the wind intensity patterns remain unchanged in these regions. Notably, this occurs when analyzing typhoons such as Tomas in 2010, Evan in 2012, and Yasa in 2020.

**Steps to Reproduce:**

- Use `TCTracks.from_ibtracs_netcdf(storm_id)` to extract tracks for historical Fiji typhoons. For instance, consider typhoons Yasa (id = 2020346S13168), Tomas (id = 2010069S12188), and Evan (id = 2012346S14180) as examples for track extraction.
- Employ the `TropCyclone.from_tracks(track, centroids, store_windfields=True)` function to derive windfields for each historical typhoon track. Specify centroids to locate the area of interest (Fiji). While optional, defining centroids allows focused analysis on the targeted region (Fiji) instead of the entire globe during windfield plotting.
- Generate wind intensity plots based on the extracted tracks using `TropCyclone.from_tracks().plot_intensity(storm_id)`.
- Observe and analyze wind intensity patterns specifically in regions where longitudes surpass 180 degrees.

Sample figures depicting wind intensity patterns for typhoons Tomas 2010, Evan 2012, and Yasa 2020 are attached below.
The issue persists across different typhoon events despite variations in their occurrence years.

**Attached Figures:**

![evan_raw_windfield](https://github.com/CLIMADA-project/climada_python/assets/54726553/234c638c-0f1a-4139-bc64-5d904cb21dfc)
![tomas_raw_windfield](https://github.com/CLIMADA-project/climada_python/assets/54726553/dd9d57ca-7b2d-468d-b07d-224cddba87a3)
![yasa_raw_windfield](https://github.com/CLIMADA-project/climada_python/assets/54726553/1d50ddd4-dc7f-44b0-99c9-574facbe5235)

Your assistance in addressing this issue would be greatly appreciated.

Thank you.

**Comment by peanutfun** [comment]
@fedemoss Thanks a lot for this report. The problem seems clearly related to the handling of coordinates at the antimeridian (which can be very tricky). You mention custom centroids you use for constructing the wind fields. Could you please provide a complete code example that provides at least one of the above plots?

**Comment by chahank** [comment]
Would it be possible to set the precomputed to `True` by default for the `from_tracks` method?

**Comment by fedemoss** [comment]
> I think that this issue is not caused by tropical cyclone code, but by the `dist_to_coast` function in `climada.util.coordinates` which seems to fail computing distances when the antimeridian is crossed. A workaround for you, @fedemoss, would be to run `centroids.set_dist_coast(precomputed=True)` on your `Centroids` object before calling `TropCyclone.from_tracks`.
> 

This actually works!! Thanks a lot. 

@peanutfun in order to recreate the plots check the .md file attached. You can get the centroids that I'm working with from [this](https://drive.google.com/file/d/1z7JrxEJTprfwirLnVea8u3-9jy3JLzGF/view?usp=drive_link) Drive folder. 

Again, thanks a lot to everyone involved.


[01.3_windfields_recreate_problem.md](https://github.com/CLIMADA-project/climada_python/files/13943057/01.3_windfields_recreate_problem.md)

**Comment by tovogt** [comment]
> Would it be possible to set the precomputed to `True` by default for the `from_tracks` method?

Yes, that's what I suggest. That basically means that `util.coordinates.dist_to_coast` will never be used anywhere in CLIMADA, and you can safely remove it.

**Comment by chahank** [comment]
Sounds good!

**Comment by tovogt** [comment]
See #840 if you agree that always using precomputed distances to coast is the way to go.

**Comment by peanutfun** [comment]
@tovogt Just for clarification: There *is* an issue with TropCyclone for storms that cross the antimeridian, but the pattern reported here occurs (exclusively?) due to issues with `dist_to_coast`?

**Comment by tovogt** [comment]
Yes, with the current develop (or main) branch and a track that crosses the antimeridian, you either run in the issues described here or in a different issue, depending on the "normalization" of the longitudinal coordinates of your centroids:
```python
from climada.hazard import Centroids, TCTracks, TropCyclone

bounds = (179.8, -17, 180.2, -16.7)
for renormalize in [False, True]:
    cen = Centroids.from_pnt_bounds(bounds, 10 / (60 * 60))
    cen.set_meta_to_lat_lon()
    
    if renormalize:
        # renormalize lon to range [-180, 180]
        cen.lon[cen.lon > 180] -= 360

    tracks = TCTracks.from_ibtracs_netcdf(storm_id=["2020346S13168"])
    wind_haz = TropCyclone.from_tracks(tracks, cen)
    
    if renormalize:
        # for plotting, renormalize lon to [0, 360]
        wind_haz.centroids.lon[wind_haz.centroids.lon < 0] += 360

    wind_haz.plot_intensity(event=0)
```
![image](https://github.com/CLIMADA-project/climada_python/assets/57705593/029d3d62-4c74-44d5-b0a5-2ed588761d3c)
My pull request #839 fixes the lower, and my pull request #840 fixes the upper issue.

**Comment by peanutfun** [comment]
Fixed by #839 #840. Thank you, @tovogt!
